src/components/carousel/types.tsx

```diff
@@ -40,7 +40,7 @@ export interface CarouselProps extends ScrollViewProps {
   /**
    * callback for when page has changed
    */
-  onChangePage?: (newPageIndex: number, oldPageIndex: number) => void;
+  onChangePage?: (newPageIndex: number, oldPageIndex: number, isAutoScrolled: boolean) => void;
```

Comment:
Consider passing the new (last) argument as an object of "info" 
If in the future we'll have to pass more info props then instead of adding more arguments, we'll just add to this info object you created.